charts/tenantfilesystem/templates/composition.yaml

```diff
@@ -0,0 +1,808 @@
+            - type: convert
+              convert:
+                toType: string
+        - fromFieldPath: "spec.product"
+          toFieldPath: "spec.forProvider.tags[4].value"
+    - name: "providerConfigRef"
+      patches:
+        - fromFieldPath: "spec.providerConfig"
+          toFieldPath: "spec.providerConfigRef.name"
+          policy:
+            fromFieldPath: Required
+
+  resources:
+    - name: TenantFileSystemEnv
+      base:
+        apiVersion: wadtfy.bmwgroup.net/v1beta1
+        kind: XTenantFileSystemEnv
+      patches:
+        - type: FromEnvironmentFieldPath
+          fromFieldPath: "iamOidcProviderId"
+          toFieldPath: "spec.env.iamOidcProviderId"
+        - type: ToCompositeFieldPath
+          fromFieldPath: "spec.env.iamOidcProviderId"
+          toFieldPath: "spec.iamOidcProviderId"
+          policy:
+            fromFieldPath: Required
+
+    - name: kms-key
+      base:
+        apiVersion: kms.aws.crossplane.io/v1alpha1
+        kind: Key
+        metadata:
+          labels:
+            crossplane/xrd-kind: xtenantfilesystem.wadtfy.bmwgroup.net
+            crossplane/xrd-helm_chart_version: {{ .Chart.Version }}
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            tags:
+              - tagKey: "wadtfy.bmwgroup.net/xrd-type"
+                tagValue: "XTenantFileSystem"
+              - tagKey: "wadtfy.bmwgroup.net/xrd-version"
+                tagValue: "v1alpha1"
+              - tagKey: "wadtfy.bmwgroup.net/component-name"
+                tagValue: ""
+              - tagKey: "wadtfy.bmwgroup.net/cluster-account-id"
+                tagValue: ""
+              - tagKey: "wadtfy.bmwgroup.net/product-name"
+                tagValue: ""
+            enableKeyRotation: true
+            enabled: true
+            policy: ""
+            region: ""
+          providerConfigRef:
+            name: ""
+      patches:
+        - type: PatchSet
+          patchSetName: "metadata"
+        - type: PatchSet
+          patchSetName: "providerConfigRef"
+        - type: PatchSet
+          patchSetName: "tags"
+        - fromFieldPath: "spec.deletePolicy"
+          toFieldPath: "spec.deletePolicy"
+        - fromFieldPath: "spec.region"
+          toFieldPath: "spec.forProvider.region"
+          policy:
+            fromFieldPath: Required
+        - fromFieldPath: "spec.accountID"
+          toFieldPath: "spec.forProvider.policy"
+          transforms:
+            - type: string
+              string:
+                fmt: |
+                  {
+                    "Version": "2012-10-17",
+                    "Id": "key-policy",
+                    "Statement": [
+                      {
+                        "Sid": "Enable IAM User Permissions",
+                        "Effect": "Allow",
+                        "Principal": {
+                          "AWS": "arn:aws:iam::%s:root"
+                        },
+                        "Action": "kms:*",
```

Comment:
not sure, but isn't that way too open as a policy? can that be further restricted to explicit kms actions?

Reply:
Well, could be the case.

Need some more tests to restrict the policy.